config/vlan.py

```diff
@@ -32,28 +32,53 @@ def is_dhcp_relay_running():
 
 
 @vlan.command('add')
-@click.argument('vid', metavar='<vid>', required=True, type=int)
+@click.argument('vid', metavar='<vid>', required=True)
+@click.option('-m', '--multiple', is_flag=True, help="Add Multiple Vlan(s) in Range or in Comma separated list")
 @clicommon.pass_db
-def add_vlan(db, vid):
+def add_vlan(db, vid, multiple):
     """Add VLAN"""
 
     ctx = click.get_current_context()
-    vlan = 'Vlan{}'.format(vid)
 
     config_db = ValidatedConfigDBConnector(db.cfgdb)
+
+    vid_list = []
+    # parser will parse the vid input if there are syntax errors it will throw error
+    if multiple:
+        vid_list = clicommon.multiple_vlan_parser(ctx, vid)
+    else:
+        if not vid.isdigit():
+            ctx.fail("{} is not integer".format(vid))
+        vid_list.append(int(vid))
+
     if ADHOC_VALIDATION:
-        if not clicommon.is_vlanid_in_range(vid):
-            ctx.fail("Invalid VLAN ID {} (1-4094)".format(vid))
 
-        if vid == 1:
-            ctx.fail("{} is default VLAN".format(vlan)) # TODO: MISSING CONSTRAINT IN YANG MODEL
+        # loop will execute till an exception occurs
+        for vid in vid_list:
+
+            if not clicommon.is_vlanid_in_range(vid):
+                ctx.fail("Invalid VLAN ID {} (2-4094)".format(vid))
+            
+            #Multiple VLANs need to be referenced
+            vlan = 'Vlan{}'.format(vid)
 
-        if clicommon.check_if_vlanid_exist(db.cfgdb, vlan): # TODO: MISSING CONSTRAINT IN YANG MODEL
-            ctx.fail("{} already exists".format(vlan))
-        if clicommon.check_if_vlanid_exist(db.cfgdb, vlan, "DHCP_RELAY"):
-            ctx.fail("DHCPv6 relay config for {} already exists".format(vlan))
-    # set dhcpv4_relay table
-    set_dhcp_relay_table('VLAN', config_db, vlan, {'vlanid': str(vid)})
+            # default vlan checker
+            if vid == 1:
+                # TODO: MISSING CONSTRAINT IN YANG MODEL
+                ctx.fail("{} is default VLAN.".format(vlan))
+
+            log.log_info("'vlan add {}' executing...".format(vid))
+
+            # TODO: MISSING CONSTRAINT IN YANG MODEL
+            if clicommon.check_if_vlanid_exist(db.cfgdb, vlan):
+                log.log_info("{} already exists".format(vlan))
+                ctx.fail("{} already exists, Aborting!!!".format(vlan))
+				
+            if clicommon.check_if_vlanid_exist(db.cfgdb, vlan, "DHCP_RELAY"):
+                ctx.fail("DHCPv6 relay config for {} already exists".format(vlan))
+
+            # set dhcpv4_relay table
+            set_dhcp_relay_table('VLAN', config_db, vlan, {'vlanid': str(vid)})
```

Comment:
Suggest either renaming the `set_dhcp_relay_table` function to something more generic (maybe `set_db_table_entry`?), or since it's a one-line function just replacing it with `config_db.set_entry('VLAN', vlan, {'vlanid': str(vid)})`

Reply:
These are exisitng VLAN & DHCP relay code and we have not modified any of this exisiting behavior of set_db_table_entry function.